src/details/MovieDetailsCard.js

```diff
@@ -57,9 +60,13 @@ class MovieDetailsCard extends Component {
   }
 
   componentDidMount() {
-    fetchMovieDetails(this.state.id)
-    .then(movieDetails => this.setState({movie: movieDetails}))
-    .catch(err => this.setState({error: this.handleErrorResponse(err.message)}))
+    if (!this.state.movie) {
+      fetchMovieDetails(this.state.id)
+      .then(movieDetails => {
+        this.setState({ movie: movieDetails })
+      })
+      .catch(err => this.setState({error: this.handleErrorResponse(err.message)}))
+    }
   }
```

Comment:
Good refactor here - while this won't eliminate this component's need to make network requests completely, it will reduce the number of network requests being made which will save us some computation.